source/common/tracing/http_tracer_impl.cc

```diff
@@ -163,6 +166,25 @@ void HttpTracerUtility::finalizeSpan(Span& span, const Http::HeaderMap* request_
   span.setTag(Tracing::Tags::get().ResponseFlags,
               StreamInfo::ResponseFlagUtils::toShortString(stream_info));
 
+  // GRPC data.
+  if (response_headers && response_trailers && stream_info.protocol() == Http::Protocol::Http2 &&
```

Comment:
Q: Don't you want to handle trailer only (header only) gRPC responses also? These would be useful error cases. cc @lizan 

/wait-any

Reply:
Right this should be extract first of non-null HeaderMap in (response trailers, response headers), I'd not limit the protocol = HTTP2 here since it is also useful cases for grpc_http1_bridge etc, and future gRPC over QUIC(HTTP/3?).